tests/integration/components/uc/network-qr-code-test.js

```diff
@@ -0,0 +1,177 @@
+import { module, test } from "qunit";
+import { setupRenderingTest } from "ember-qunit";
+import { render, triggerEvent, click } from "@ember/test-helpers";
+import Service from "@ember/service";
+import { hbs } from "ember-cli-htmlbars";
+
+class MockQrCodeService extends Service {
+  toCanvas() {}
+
+  toDataURL() {
+    return "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABAQMAAAAl21bKAAAAA1BMVEUAAACnej3aAAAAAXRSTlMAQObYZgAAAApJREFUCNdjYAAAAAIAAeIhvDMAAAAASUVORK5CYII="; // 1x1 pixel white
+  }
+}
+
+module("Integration | Component | uc/network-qr-code", function (hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function () {
+    this.owner.register("service:qr-code", MockQrCodeService);
+  });
+
+  test("it renders a canvas with the QR code", async function (assert) {
+    await render(
+      hbs`<Uc::NetworkQrCode @ssid="mynetwork" @password="mypass" @encryption="wpa" />`
+    );
+
+    assert.dom("[data-test-canvas]").exists();
+  });
+
+  test("it draws the QR code when rendering", async function (assert) {
+    this.owner.register(
```

Comment:
I think it's a good idea to declare how many assertions we expect to be fired, especially in 'async' cases
```suggestion
  test("it draws the QR code when rendering", async function (assert) {
    assert.expect(1);
    this.owner.register(
```